src/Command/ListCollections.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/*
+ * Copyright 2015-2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Command;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Driver\Session;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Model\CachingIterator;
+use MongoDB\Operation\Executable;
+use function is_array;
+use function is_bool;
+use function is_integer;
+use function is_object;
+
+/**
+ * Wrapper for the listCollections command.
+ *
+ * @internal
+ * @see http://docs.mongodb.org/manual/reference/command/listCollections/
+ */
+class ListCollections implements Executable
```

Comment:
The main advantage of using `nameOnly` is that the server does not need to lock each collection in the database, which it needs to do when grabbing full statistics. In the interest of not creating `CollectionInfo` documents with partial information, I opted against re-using the `ListCollection` operation even when we don't expose the custom iterator.

Another main reason for extracting to its own namespace was to avoid conflicts between the command and the `ListCollections` operation, which does not accept the `nameOnly` option due to the reason above.

Reply:
Works for me.